src/model/impl/TaskManager.cpp

```diff
@@ -18,7 +18,14 @@ TaskManager::TaskManager(std::unique_ptr<IdGenerator> generator)
 
 ModelResponse TaskManager::Add(const Task& task)
 {
-    TaskId new_id = gen_->GetNextId();
+    TaskId new_id;
+    {
+        std::lock_guard<std::mutex> lock{id_generation_mutex_};
```

Comment:
generally, since you're protecting generator outside of the generator, you have no guarantee noone would call it without the mutex protection.
This will work, since you have a unique instance of a generator incapsulated in your class. Works fine, but not obvious choice.

Reply:
Fixed